sentry-sidekiq/spec/sentry/sidekiq/cron/job_spec.rb

```diff
@@ -79,4 +92,74 @@
+      it 'adds job to sidekiq within transaction' do
+        job = Sidekiq::Cron::Job.new(name: 'test', cron: 'not a crontab', class: 'HappyWorkerForCron')
+        job.enque!
+
+        expect(::Sidekiq::Queue.new.size).to eq(1)
+        expect(transport.events.count).to eq(1)
+        event = transport.events.last
+        expect(event.spans.count).to eq(1)
+        expect(event.spans[0][:op]).to eq("queue.publish")
+        expect(event.spans[0][:data]['messaging.destination.name']).to eq('default')
+      end
+
+      it 'adds job to sidekiq within transaction' do
+        job = Sidekiq::Cron::Job.new(name: 'test', cron: 'not a crontab', class: 'HappyWorkerForCron')
+        job.enque!
+        # Time passes.
+        job.enque!
+
+        expect(::Sidekiq::Queue.new.size).to eq(2)
+        expect(transport.events.count).to eq(2)
+        events = transport.events
+        expect(events[0].spans.count).to eq(1)
+        expect(events[0].spans[0][:op]).to eq("queue.publish")
+        expect(events[0].spans[0][:data]['messaging.destination.name']).to eq('default')
+        expect(events[1].spans.count).to eq(1)
+        expect(events[1].spans[0][:op]).to eq("queue.publish")
+        expect(events[1].spans[0][:data]['messaging.destination.name']).to eq('default')
+
+        expect(events[0].dynamic_sampling_context['trace_id']).to_not eq(events[1].dynamic_sampling_context['trace_id'])
+      end
+    end
+  end
+
+  if Gem.loaded_specs["sidekiq-cron"].version >= Gem::Version.new('2.0.0')
+    describe "sidekiq-cron >= 2.0.0" do
+      it 'adds job to sidekiq within transaction' do
+        job = Sidekiq::Cron::Job.new(name: 'test', cron: 'not a crontab', class: 'HappyWorkerForCron')
+        job.enqueue!
+
+        expect(::Sidekiq::Queue.new.size).to eq(1)
+        expect(transport.events.count).to eq(1)
+        event = transport.events.last
+        expect(event.spans.count).to eq(1)
+        expect(event.spans[0][:op]).to eq("queue.publish")
+        expect(event.spans[0][:data]['messaging.destination.name']).to eq('default')
+      end
+
+      it 'adds job to sidekiq within transaction' do
+        job = Sidekiq::Cron::Job.new(name: 'test', cron: 'not a crontab', class: 'HappyWorkerForCron')
+        job.enqueue!
+        # Time passes.
+        job.enqueue!
+
+        expect(::Sidekiq::Queue.new.size).to eq(2)
+        expect(transport.events.count).to eq(2)
+        events = transport.events
+        expect(events[0].spans.count).to eq(1)
+        expect(events[0].spans[0][:op]).to eq("queue.publish")
+        expect(events[0].spans[0][:data]['messaging.destination.name']).to eq('default')
+        expect(events[1].spans.count).to eq(1)
+        expect(events[1].spans[0][:op]).to eq("queue.publish")
+        expect(events[1].spans[0][:data]['messaging.destination.name']).to eq('default')
+
+        expect(events[0].dynamic_sampling_context['trace_id']).to_not eq(events[1].dynamic_sampling_context['trace_id'])
+      end
+    end
```

Comment:
It would be good to have `Sentry::Sidekiq::Cron.enqueueing_method` that returns either `:enque!` or `:enqueue!` and then remove these conditionals and just do `job.public_send(Sentry::Sidekiq::Cron.enqueueing_method)`.